app/controllers/spree/admin/invoices_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Admin
+    class InvoicesController < Spree::Admin::BaseController
+      respond_to :json
+
+      def create
+        Delayed::Job.enqueue BulkInvoiceJob.new(params[:order_ids], directory, filename)
+
+        render text: filename, status: :ok
+      end
+
+      def show
+        invoice_id = params[:id]
+
+        send_file(filepath(invoice_id), type: 'application/pdf', disposition: :inline)
+      end
+
+      def poll
+        invoice_id = params[:invoice_id]
```

Comment:
That was how it came out in the generated routes. 

It looks like the solution involves waiting for Rails 4, or using an extra gem: https://stackoverflow.com/questions/6592038/change-the-name-of-the-id-parameter-in-routing-resources-for-rails

Reply:
I get it, it's because it considers "poll" to be another resource other than invoice and so the invoice id gets named "invoice_id" in the poll resource.